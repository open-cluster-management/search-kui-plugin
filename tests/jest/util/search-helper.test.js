/*******************************************************************************
*
* Copyright (c) 2020 Red Hat, Inc.
*
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
'use strict'
const helper = require('../../../dist/util/search-helper')

describe('Search helper', () => {
  const commands = [
    'search kind:pod,deployment namespace:kube-system',
    'search summary kind:pod name:nvidia-device-plugin-r2674 namespace:kube-system',
    'search kind:pod --related',
    'search pod,deployment',
    'search -h',
    'search completions:=1'
  ]
  commands.forEach(command => {
    const spy = jest.spyOn(helper, 'convertStringToQuery')
    const _ = helper.convertStringToQuery(command)

    it(`should convert ${command} to a query`, () => {
      expect(spy).toBeCalled()
      expect(helper.convertStringToQuery(command)).not.toBeUndefined()

      expect(helper.convertStringToQuery(command)).toEqual(
        expect.objectContaining({
          filters: expect.any(Array),
          keywords: expect.any(Array)
        })
      )

      expect(helper.convertStringToQuery(command)).toMatchSnapshot()
    })
  });
})
