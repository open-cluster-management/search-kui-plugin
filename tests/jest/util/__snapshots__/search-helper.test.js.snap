// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search helper should convert search -h to a query 1`] = `
Object {
  "filters": Array [],
  "keywords": Array [
    "-h",
  ],
}
`;

exports[`Search helper should convert search completions:=1 to a query 1`] = `
Object {
  "filters": Array [
    Object {
      "property": "completions",
      "values": Array [
        "=1",
      ],
    },
  ],
  "keywords": Array [],
}
`;

exports[`Search helper should convert search kind:pod --related to a query 1`] = `
Object {
  "filters": Array [
    Object {
      "property": "kind",
      "values": Array [
        "pod",
      ],
    },
  ],
  "keywords": Array [],
}
`;

exports[`Search helper should convert search kind:pod,deployment namespace:kube-system to a query 1`] = `
Object {
  "filters": Array [
    Object {
      "property": "kind",
      "values": Array [
        "pod",
        "deployment",
      ],
    },
    Object {
      "property": "namespace",
      "values": Array [
        "kube-system",
      ],
    },
  ],
  "keywords": Array [],
}
`;

exports[`Search helper should convert search pod,deployment to a query 1`] = `
Object {
  "filters": Array [],
  "keywords": Array [
    "pod,deployment",
  ],
}
`;

exports[`Search helper should convert search summary kind:pod name:nvidia-device-plugin-r2674 namespace:kube-system to a query 1`] = `
Object {
  "filters": Array [
    Object {
      "property": "kind",
      "values": Array [
        "pod",
      ],
    },
    Object {
      "property": "name",
      "values": Array [
        "nvidia-device-plugin-r2674",
      ],
    },
    Object {
      "property": "namespace",
      "values": Array [
        "kube-system",
      ],
    },
  ],
  "keywords": Array [],
}
`;
